src/components/PeopleTable/PeopleTable.tsx

```diff
@@ -0,0 +1,176 @@
+import { useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { Person } from '../../types';
+import { NoPeopleMessage } from '../NoPeopleMessage/NoPeopleMessage';
+import { PersonLink } from '../PersonLink/PersonLink';
+import { Sex } from '../../types/Sex';
+import { SearchLink } from '../SearchLink';
+import { SortCategory } from '../../types/sortCategory';
+import { handleSort } from '../../helpers/handleSort';
+import { getSortIconClass } from '../../helpers/getSortIconClass';
+
+type Props = {
+  people: Person[];
+};
+
+export const PeopleTable: React.FC<Props> = ({ people }) => {
+  const [filteredPeople, setFilteredPeople] = useState<Person[]>(people);
+
+  const [searchParams] = useSearchParams();
+
+  useEffect(() => {
+    let filteredList = [...people];
+    const query = searchParams.get('query') || '';
+    const centuries = searchParams.getAll('centuries') || [];
+
+    const lowerCaseQuery = query.toLowerCase().trim();
+
+    filteredList = people.filter(
+      (person) => person.name.toLowerCase().includes(lowerCaseQuery)
+        || person.motherName?.toLowerCase().includes(lowerCaseQuery)
+        || person.fatherName?.toLowerCase().includes(lowerCaseQuery),
```

Comment:
this is repeating, create simple reusable function for optimizing case and checking for include